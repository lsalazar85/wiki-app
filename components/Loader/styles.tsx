import styled, {keyframes, css} from "styled-components";
import Image from "next/image";

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const animationStyle = css`
  animation: ${rotate} 2s linear infinite;
`;

export const LoaderContainer = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`

export const LoaderImage = styled(Image)`
  ${animationStyle}
`
