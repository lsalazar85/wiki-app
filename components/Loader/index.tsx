import {LoaderContainer, LoaderImage } from './styles'

const Loader = () => (
    <LoaderContainer>
      <LoaderImage
          src="/images/rick.png"
          alt="Loader Icon"
          layout="intrinsic"
          width={90}
          height={90}
      />
    </LoaderContainer>
)

export default Loader
