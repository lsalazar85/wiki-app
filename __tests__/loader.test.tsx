/**
 * @jest-environment jsdom
 */

import { render  } from '@testing-library/react'
import Loader from "../components/Loader";

describe('Loader', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<Loader />)

        expect(asFragment()).toBeDefined()
    })
})
