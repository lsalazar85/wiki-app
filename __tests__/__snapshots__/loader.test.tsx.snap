// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render successfully 1`] = `
<DocumentFragment>
  <div
    class="styles__LoaderContainer-sc-1sn62nv-0 btuLRX"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iOTAiIGhlaWdodD0iOTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="Loader Icon"
        class="styles__LoaderImage-sc-1sn62nv-1 fwBpdA"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </div>
  </div>
</DocumentFragment>
`;
